src/utils/monaco/highlightErrors.ts

```diff
@@ -0,0 +1,47 @@
+import {parseYqlQueryWithoutCursor} from '@gravity-ui/websql-autocomplete/yql';
+import {MarkerSeverity, editor} from 'monaco-editor';
+
+import i18n from './i18n';
+
+const owner = 'ydb';
+
+let errorsHighlightingTimeoutId: ReturnType<typeof setTimeout>;
+
+export function updateErrorsHighlighting() {
+    unHighlightErrors();
+
+    clearTimeout(errorsHighlightingTimeoutId);
+    errorsHighlightingTimeoutId = setTimeout(() => highlightErrors(), 500);
+}
+
+function highlightErrors() {
+    const model = window.ydbEditor?.getModel();
+    if (!model) {
+        console.error('unable to retrieve model when highlighting errors');
+        return;
+    }
+
+    const errors = parseYqlQueryWithoutCursor(model.getValue()).errors;
+    if (!errors.length) {
+        unHighlightErrors();
+        return;
+    }
+
+    const markers = errors.map((error): editor.IMarkerData => {
+        const markerColumn = error.startColumn + 1;
+        return {
+            message: i18n('context_syntax-error'),
+            source: error.message,
+            severity: MarkerSeverity.Error,
+            startLineNumber: error.startLine,
+            startColumn: markerColumn,
+            endLineNumber: error.endLine,
+            endColumn: markerColumn,
```

Comment:
looks like start and end column are the same - is it expected?

Reply:
omg, I'm so ashamed :(